### PR TITLE
Adds carbondb

### PR DESCRIPTION
**Pull request by ribalba**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -35,6 +39,19 @@ inputs:
       description: 'Base URL of the Github API to send data to. Default is api.github.com, but can be changed to your hostname if you have Github Enterprise'
       default: 'api.github.com'
       required: false
+  company-uuid:
+      description: 'If you want to add data to the CarbonDB you can set this to your company UUID'
+      default: ''
```
Comment (ArneTR) [inline action.yml:44]: The other optional arguments like `label` are default null. Is it possible to unify that to either?

File: README.md [hunk 2]
```diff
@@ -77,35 +77,46 @@ jobs:
...
 - `label`: (optional) (default: 'measurement ##')
-    - Used with `get_measurement` and `display_results` to identify the measurement
+  - Used with `get_measurement` and `display_results` to identify the measurement
 - `send-data`: (optional) (default: true)
-    - Send metrics data to metrics.green-coding.io to create and display badge, and see an overview of the energy of your CI runs. Set to false to send no data. The data we send are: the energy value and duration of measurement; cpu model; repository name/branch/workflow_id/run_id; commit_hash; source (GitHub or GitLab). We use this data to display in our green-metrics-tool front-end here: https://metrics.green-coding.io/ci-index.html 
+  - Send metrics data to metrics.green-coding.io to create and display badge, and see an overview of the energy of your CI runs. Set to false to send no data. The data we send are: the energy value and duration of measurement; cpu model; repository name/branch/workflow_id/run_id; commit_hash; source (GitHub or GitLab). We use this data to display in our green-metrics-tool front-end here: https://metrics.green-coding.io/ci-index.html
+- `show-carbon`: (optional) (default: true)
```
Comment (ArneTR) [inline README.md:94]: Having this default true is nice, but how will this work without the electricitymaps token?

**Review by ArneTR — CHANGES_REQUESTED** [review]

File: scripts/make_measurement.sh [hunk 3]
```diff
@@ -82,15 +82,19 @@ function make_measurement() {
         value_to_add="label:\"$label\", cpu_avg:$cpu_avg, total_energy:$total_energy, power_avg:$power_avg, time:$time"
         source "$(dirname "$0")/vars.sh" add_var $key_to_add "$value_to_add"
 
-        echo $total_energy >> /tmp/eco-ci/energy-values.txt 
+        echo $total_energy >> /tmp/eco-ci/energy-values.txt
         source "$(dirname "$0")/vars.sh" add_var MEASUREMENT_RAN true
 
+        if [ -z "$cb_machine_uuid" ]; then
+             cb_machine_uuid=$(uuidgen)
```
[the remark this answers is not in the document]
Comment (ribalba) [reply scripts/make_measurement.sh:89]:
```
The uuidgen command generates a Universally Unique IDentifier (UUID), a 128-bit value guaranteed to be unique over both space and time.
```

File: scripts/vars.sh [hunk 4]
```diff
@@ -168,6 +168,69 @@ function cpu_vars_fill {
...
+        return 1
+    fi
+
+    echo "$response"
+}
+
+get_carbon_intensity() {
+    latitude=$1
+    longitude=$2
+    token="your_electricity_maps_token_here"
```
[the remark this answers is not in the document]
Comment (ribalba) [reply scripts/vars.sh:186]: Never, as electricity maps works without a token. The string is not chose very well. 
Added the option for a var

File: scripts/vars.sh [hunk 5]
```diff
@@ -168,6 +168,69 @@ function cpu_vars_fill {
...
+    if [ "$token" == "testing" ]; then
+        echo 1000
+        return
+    fi
+
+    response=$(curl -s -H "auth-token: $token" "https://api.electricitymap.org/v3/carbon-intensity/latest?lat=$latitude&lon=$longitude")
+
+    if echo "$response" | jq '.carbonIntensity' | grep -q null; then
+        echo "Required carbonIntensity is missing. Exiting" >&2
+        return 1
```
[the remark this answers is not in the document]
Comment (ribalba) [reply scripts/vars.sh:197]: 1 is for error. The check for empty string was from before I wrote stuff the `stderr`

File: action.yml (hunk 1 above)
Comment (ribalba) [reply action.yml:44]: I used empty string here as I can just pass it into the curl post command and not really care as the backend will then ignore it. I am not sure if handling null will make it a lot cleaner

**Comment by ArneTR** [comment]
Thanks for the clarification that the token is optional.

I am not a fan however of the multiple ways to submit input. All other info is supplied via arguments. Only the Electricitymaps Token is now an exported environment var.

What is the rationale behind this?

In any case: I would favor for supplying it as argument and also making it a secret by default. Not a classic ENV.

2. These two values submitted in the PR info need a dimension.

Carbon Intensity for this location: 99 (g / kg / tons per Wh, J , kwH ?)
CO2eq emitted for this job: 0.002452 (g, kg, ton ?)

**Comment by ribalba** [comment]
The problem is that when we do it with a secret/ not as env we will need to add it as a parameter to the `display_results.sh` and then it will be visible in the logs. Which is not ideal.

**Comment by ArneTR** [comment]
TBD tomorrow

**Comment by ribalba** [comment]
@ArneTR 

- [x] Add error text when apis fail and not fail silently
- [x] Curl will stop the script if it fails. Add detailed error checking
- [x] Token as secret

**Pull request merged**